src/server/service/kv.rs

```diff
@@ -1055,6 +708,805 @@ impl<T: RaftStoreRouter + 'static, E: Engine> tikvpb_grpc::Tikv for Service<T, E
+        }
+        Some(BatchCommandsRequest_Request_oneof_cmd::RawPut(req)) => {
+            let resp = future_raw_put(&storage, req)
+                .map(BatchCommandsResponse_Response_oneof_cmd::RawPut)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.raw_put.inc());
+            response_batch_commands_request(executor, id, resp, tx);
+        }
+        Some(BatchCommandsRequest_Request_oneof_cmd::RawBatchPut(req)) => {
+            let resp = future_raw_batch_put(&storage, req)
+                .map(BatchCommandsResponse_Response_oneof_cmd::RawBatchPut)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.raw_batch_put.inc());
+            response_batch_commands_request(executor, id, resp, tx);
+        }
+        Some(BatchCommandsRequest_Request_oneof_cmd::RawDelete(req)) => {
+            let resp = future_raw_delete(&storage, req)
+                .map(BatchCommandsResponse_Response_oneof_cmd::RawDelete)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.raw_delete.inc());
+            response_batch_commands_request(executor, id, resp, tx);
+        }
+        Some(BatchCommandsRequest_Request_oneof_cmd::RawBatchDelete(req)) => {
+            let resp = future_raw_batch_delete(&storage, req)
+                .map(BatchCommandsResponse_Response_oneof_cmd::RawBatchDelete)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.raw_batch_delete.inc());
+            response_batch_commands_request(executor, id, resp, tx);
+        }
+        Some(BatchCommandsRequest_Request_oneof_cmd::RawScan(req)) => {
+            let resp = future_raw_scan(&storage, req)
+                .map(BatchCommandsResponse_Response_oneof_cmd::RawScan)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.raw_scan.inc());
+            response_batch_commands_request(executor, id, resp, tx);
+        }
+        Some(BatchCommandsRequest_Request_oneof_cmd::RawDeleteRange(req)) => {
+            let resp = future_raw_delete_range(&storage, req)
+                .map(BatchCommandsResponse_Response_oneof_cmd::RawDeleteRange)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.raw_delete_range.inc());
+            response_batch_commands_request(executor, id, resp, tx);
+        }
+        Some(BatchCommandsRequest_Request_oneof_cmd::RawBatchScan(req)) => {
+            let resp = future_raw_batch_scan(&storage, req)
+                .map(BatchCommandsResponse_Response_oneof_cmd::RawBatchScan)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.raw_batch_scan.inc());
+            response_batch_commands_request(executor, id, resp, tx);
+        }
+        Some(BatchCommandsRequest_Request_oneof_cmd::Coprocessor(req)) => {
+            let resp = future_cop(&cop, req, Some(peer))
+                .map(BatchCommandsResponse_Response_oneof_cmd::Coprocessor)
+                .map_err(|_| GRPC_MSG_FAIL_COUNTER.coprocessor.inc());
+            response_batch_commands_request(executor, id, resp, tx);
+        }
+        None => panic!("invalid super batch request"),
```

Comment:
Please don't panic, since invalid requests are possible, for example, outdated clients, other clients sending wrong message to the port, etc.